File: 70_Climbing_Stairs/Solution.cpp
class Solution {
public:
    int memo[46];
    int solve(int n){
        if(n<0){
            return 0;
        }
        if(n==0){
            return 1;
        }
        if(memo[n]!=-1){
            return memo[n];
        }
        int take1Step = solve(n-1);
        int take2Steps = solve(n-2);
        return memo[n] = take1Step + take2Steps;
    }
    int climbStairs(int n) {
        memset(memo,-1,sizeof(memo));
        return solve(n);
    }
};

File: 70_Climbing_Stairs/Solution.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 1 or n == 2:
            return n

        a = 1  # i-2
        b = 2  # i-1
        c = 3  # i

        for i in range(3, n + 1):
            c = a + b
            temp_b = b
            b = c
            a = temp_b

        return c
